### PR TITLE
Add in upstream changes from v8eval

### PR DESCRIPTION
**Pull request by paetling**

### CODE REVIEW

**Comment by paetling** [comment]
review @confucious 

**Context** 
We were seeing the following bug when trying to build our servertooth image based off of the servertooth_base image. 

```
/go/src/github.com/gamechanger/v8eval/go/v8eval/v8.go:40: undefined: X_GoV8
```

This seemed to imply to me that the v8 c bindings were not properly install in the base image. (We had to roll a new base image for servertooth found here: https://github.com/gamechanger/docker-images/tree/master/servertooth because travis' credentials used to be stored in our base image). 

I attempted to rebuild out base image and did in fact find that the image was not building properly. I was seeing issues with not finding the v8 binarys that were needed to link to build the v8eval go package. 

Digging deeper I discovered that the it seemed like running make in v8 was not working. 

With all of this in mind, I wanted to look at upstream changes that have happened to [v8eval](https://github.com/sony/v8eval). It seems like there were some issues fixed that dealt with similar errors to what I was seeing: https://github.com/sony/v8eval/issues/27. 

this pr fixes are issue by grabbing all upstream changes to the sony v8eval and adding them to our fork. This seems to fix the base image build issue and subsequently our servertooth image build issue.

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.5.1
+FROM golang:1.8.3
```
Comment (paetling) [inline Dockerfile:1]: we do not actually use this dockerfile so this golang version does not really matter

File: build.sh [hunk 2]
```diff
@@ -59,11 +59,12 @@ install_v8() {
   cd $V8EVAL_ROOT
   fetch v8
   cd v8
-  git checkout 5.2.163
+  git checkout 6.3.288
```
Comment (paetling) [inline build.sh:62]: we are using a higher version of v8. I am checking the change log for v8 now.

Comment (paetling) [reply build.sh:62]: * https://v8project.blogspot.com/2016/12/v8-release-56.html
* https://v8project.blogspot.com/2017/04/v8-release-59.html
* https://v8project.blogspot.com/2017/08/v8-release-61.html
* https://v8project.blogspot.com/2017/10/v8-release-63.html

These 4 seem like they introduce some improvements that might speed up our system or reduce our memory consumption, but nothing seems to scary with respect to changes. All of these just seem like improvements

It is still probably worth testing this new version of servertooth on this version of v8 on staging first

File: Dockerfile (hunk 1 above)
Comment (confucious) [reply Dockerfile:1]: Should we remove this dockerfile? Is it used for anything at all? I want to make sure we don't confuse someone looking at the project as to where the environment comes from.

Comment (confucious) [reply Dockerfile:1]: Oh, I see. this is v8eval. So these are upstream changes. The Dockerfile I'm thinking about is the one in servertooth itself.

**Comment by confucious** [comment]
Seems to make sense. Looks like the main change is that we're switching to 6.3 v8. It looks like most of the other changes are around diagnostics and unit tests. I can handle the servertooth deploy if you want to merge this and kick off the docker build environment.

File: Dockerfile (hunk 1 above)
Comment (paetling) [reply Dockerfile:1]: Yes

**Pull request merged**